RELEASE.md

```diff
@@ -1,9 +1,42 @@
-# Release Process
+# Release Guide
 
-The Kubernetes Template Project is released on an as-needed basis. The process is as follows:
+## Semi-automatic
 
-1. An issue is proposing a new release with a changelog since the last release
```

Comment:
Usually we have one issue showing the todos/progress of a release, so let's keep this step. [An example](https://github.com/kubernetes-sigs/scheduler-plugins/issues/330).